# HG changeset patch
# User Phil Pennock <mutt-dev@spodhuis.demon.nl>
# Date 1454170933 0
#      Sat Jan 30 16:22:13 2016 +0000
# Branch gentoo-1.7
# Node ID 32b1a18320f46959497d3ac38711cef7cdf72393
# Parent  08325b7e2ef8858fd3e6ff0e0b9d66972c7f0118
[PATCH] feature: tls-sni
From 7b1b46f67e08f96d2853587ead4368139875b62d Mon Sep 17 00:00:00 2001
Negotiate with a server for a TSL/SSL certificate
---
 mutt_ssl.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff -r 08325b7e2ef8 -r 32b1a18320f4 PATCHES
--- a/PATCHES	Mon Jan 25 17:02:57 2016 +0000
+++ b/PATCHES	Sat Jan 30 16:22:13 2016 +0000
@@ -1,3 +1,4 @@
+patch-tls-sni-neomutt
 patch-index-color-neomutt
 patch-status-color-neomutt
 patch-skip-quoted-neomutt
diff -r 08325b7e2ef8 -r 32b1a18320f4 mutt_ssl.c
--- a/mutt_ssl.c	Mon Jan 25 17:02:57 2016 +0000
+++ b/mutt_ssl.c	Sat Jan 30 16:22:13 2016 +0000
@@ -412,6 +412,18 @@ static int ssl_negotiate (CONNECTION *co
   SSL_set_mode (ssldata->ssl, SSL_MODE_AUTO_RETRY);
 #endif
 
+#if (OPENSSL_VERSION_NUMBER >= 0x0090806fL) && !defined(OPENSSL_NO_TLSEXT)
+  /* TLS Virtual-hosting requires that the server present the correct
+   * certificate; to do this, the ServerNameIndication TLS extension is used.
+   * If TLS is negotiated, and OpenSSL is recent enough that it might have
+   * support, and support was enabled when OpenSSL was built, mutt supports
+   * sending the hostname we think we're connecting to, so a server can send
+   * back the correct certificate.
+   * This has been tested over SMTP against Exim 4.80.
+   * Not yet found an IMAP server which supports this. */
+  SSL_set_tlsext_host_name (ssldata->ssl, conn->account.host);
+#endif
+
   if ((err = SSL_connect (ssldata->ssl)) != 1)
   {
     switch (SSL_get_error (ssldata->ssl, err))
